### ISSUE TITLE
Mangled header when cleaning code using codemaid & resharper

### ISSUE THREAD

**Issue by interSAP** [codecadwallader/codemaid#381]
### Environment
- Visual Studio version: Microsoft Visual Studio Ultimate 2013
- CodeMaid version: 10.1
- Resharper version: Ultimate 2016.3.1
- Code language: C#

### Description
While cleaning up, for some reason the combination of code cleanup from CodeMaid and Resharper mangle the file header. The new lines are not reproduced. Using Resharper cleanup does not produce the same error.

### Steps to recreate
1.- Settings
Header defined in Resharper:
* ********************************************************************************
 (c) $CREATED_YEAR$ - blablabla.

 Solution......: $SOLUTION$
 Project.......: $PROJECT$
 Filename......: $FILENAME$

Options set in Codemaid: 
CLEANING->UPDATE
FileHeader is blank
Update accessors is blank
Update #endregion is set
Update single line is set
THIRD PARTY
Run Jetbrains ReSharper is set
2.- Launch Codemaid cleanup, either from the context menu, menu, shortcut, for one file or for the whole solution.

### Current behavior
This is the header that gets inserted:
// * ******************************************************************************** (c) 2016 - blablabla
//
// Solution......: mysolution Project.......: myproject Filename......: myfile.cs

### Expected behavior
This is the expected header:
// * ********************************************************************************
//  (c) 2016 - blablabla
// 
//  Solution......: mysolution
//  Project.......: myproject
//  Filename......: myfile.cs
//

**Comment by interSAP** [comment]
Hello, 

Setting up codemaid not to use third parties, then cleaning up mangles the header:
// * ******************************************************************************** (c) 2016 - blablabla
//
// Solution......: mysolution Project.......: myproject Filename......: myfile.cs
Running reshaper code cleanup afterwards restores the header
// * ********************************************************************************
// (c) 2016 - blablabla
//
// Solution......: mysolution
// Project.......: myproject
// Filename......: myfile.cs
//

I've tested with CodeMaid 10.2 and the issue persists.

Thanks,

**Comment by codecadwallader** [comment]
Thanks for the information.  Do you have comment formatting on save enabled?

CodeMaid->Options->Formatting->"Run format comments during cleanup"

This option is off by default, but when enabled will tell CodeMaid to try and re-format comments.  Comment formatting will consolidate comments and enforce line wrapping at a specified column.  If you do not want your comment to be modified by CodeMaid please turn this option back off and I believe you will be set.

**Comment by interSAP** [comment]
yes, was active, clearing the option seems to do the work. Still We then lose the capability to reformat other type of comments. I think "header" comments should be considered as a special case.

thanks

**Comment by codecadwallader** [comment]
Ok, converting this into an enhancement request for anyone who may want to run with it.

**Comment by adamjstone** [comment]
I'll take it, probably. It looks like the change should go near line 226 of `CodeReorganizationManager`.

Thanks, sir.

**Comment by willemduncan** [comment]
The header comment is updated because it does not match the required comment style. There is an attempt to make a list:

```
Solution......: $SOLUTION$
Project.......: $PROJECT$
Filename......: $FILENAME$
```
but the "list style" is not recognized and thus treated as being regular words and put into a single line. Changing the header as follows would work:
```
Solution: $SOLUTION$
Project:  $PROJECT$
Filename: $FILENAME$
```

I see no downside in updating the recognized list styles in the formatter to allow punctuation marks in the list prefixes though. Right now I think it only accepts word characters (A-Z and numbers).
